lazy/src/component.rs

```diff
@@ -56,36 +63,48 @@ where
             StateBuilder {
                 component: Box::new(component),
                 message: PhantomData,
-                cache_builder: |state| {
-                    Some(
-                        CacheBuilder {
-                            element: state.view(),
-                            overlay_builder: |_| None,
-                        }
-                        .build(),
-                    )
-                },
+                state: PhantomData,
+                element_builder: |_| None,
```

Comment:
I'm getting a panic using this branch since the view isn't initialized here and `layout` is called before other methods that rebuild it.

Reply:
This appears to be the same code from before, did the order in which widget methods are called get changed?

Reply:
Relevant backtrace:

```sh
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/lazy/src/component.rs:123:60
stack backtrace:
   0: rust_begin_unwind
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/panicking.rs:48:5
   3: core::option::Option<T>::unwrap
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/option.rs:755:21
   4: iced_lazy::component::Instance<Message,Renderer,Event,S>::with_element_mut::{{closure}}
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/lazy/src/component.rs:123:43
   5: iced_lazy::component::ouroboros_impl_state::State<Message,Renderer,Event,S>::with_element_mut
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/lazy/src/component.rs:78:1
   6: iced_lazy::component::Instance<Message,Renderer,Event,S>::with_element_mut
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/lazy/src/component.rs:119:9
   7: iced_lazy::component::Instance<Message,Renderer,Event,S>::with_element
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/lazy/src/component.rs:112:9
   8: <iced_lazy::component::Instance<Message,Renderer,Event,S> as iced_native::widget::Widget<Message,Renderer>>::width
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/lazy/src/component.rs:155:9
   9: iced_native::layout::flex::resolve
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/native/src/layout/flex.rs:112:33
  10: <iced_native::widget::row::Row<Message,Renderer> as iced_native::widget::Widget<Message,Renderer>>::layout
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/native/src/widget/row.rs:122:9
  11: <iced_native::widget::container::Container<Message,Renderer> as iced_native::widget::Widget<Message,Renderer>>::layout::{{closure}}
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/native/src/widget/container.rs:163:17
  12: iced_native::widget::container::layout
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/native/src/widget/container.rs:299:23
  13: <iced_native::widget::container::Container<Message,Renderer> as iced_native::widget::Widget<Message,Renderer>>::layout
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/native/src/widget/container.rs:152:9
  14: <iced_graphics::renderer::Renderer<B,T> as iced_native::renderer::Renderer>::layout
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/graphics/src/renderer.rs:61:22
  15: iced_native::user_interface::UserInterface<Message,Renderer>::build
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/native/src/user_interface.rs:99:13
  16: iced_winit::application::build_user_interface
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/winit/src/application.rs:539:26
  17: iced_glutin::application::run_instance::{{closure}}
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/glutin/src/application.rs:221:27
  18: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/future/mod.rs:91:19
  19: iced_glutin::application::run::{{closure}}
             at /Users/tarkah/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/f294c2d/glutin/src/application.rs:168:24
```

Reply:
Can confirm this is reproducible on the `component` example

Reply:
Good catch! Fixed by c23ed7e4a0a2b62a0d7cabe6e35d7323eac543d2.

Reply:
That fixed it, thanks!